backends-velox/src/main/scala/io/glutenproject/execution/VeloxColumnarToRowExec.scala

```diff
@@ -98,26 +98,13 @@ object VeloxColumnarToRowExec {
 
     // TODO:: pass the jni jniWrapper and arrowSchema  and serializeSchema method by broadcast
     val jniWrapper = NativeColumnarToRowJniWrapper.create()
-    var closed = false
     val c2rId = jniWrapper.nativeColumnarToRowInit(
       NativeMemoryManagers.contextInstance("ColumnarToRow").getNativeInstanceHandle)
 
-    TaskResources.addRecycler(s"ColumnarToRow_$c2rId", 100) {
-      if (!closed) {
-        jniWrapper.nativeClose(c2rId)
-        closed = true
-      }
-    }
-
     val res: Iterator[Iterator[InternalRow]] = new Iterator[Iterator[InternalRow]] {
 
       override def hasNext: Boolean = {
-        val hasNext = batches.hasNext
-        if (!hasNext && !closed) {
-          jniWrapper.nativeClose(c2rId)
-          closed = true
-        }
-        hasNext
+        batches.hasNext
```

Comment:
It seems we didi not call `recyclePayload` for c2r after refactor since the iterator is row-based. Is there a leak ?

Reply:
Didn't change any code about payload closing. The deleted lines were about iterator completion so moved to`recycleIterator`.

Reply:
I see it is closed by itself, then it seems c2r did not call `recyclePayload` ?

Reply:
> I see it is closed by itself, then it seems c2r did not call `recyclePayload` ?

Yes as you see the batches are closed manually within some `if-else` conditions. I don't how much effort needed to refactor the usages within `#recyclePayload()` so didn't change that part of code in this patch. Probably we can do that in a separate ticket, I am not sure.